### PR TITLE
Changes to fix multiple definition error when using musl

### PR DESCRIPTION
**Pull request by WorksButNotTested**

### CODE REVIEW

**Comment by WorksButNotTested** [comment]
I had previously been working with just frida-gum and injecting the frida-gadget.so using my own tools or via LD_PRELOAD, now that I have hopefully addressed the issues I was finding with frida-inject (both related to the auxilliary vector) I am hoping to move to just using FRIDA out of the box.

Hence I have now tried building the core with musl and uclibc. Then I stumbled across this really weird issue. Hope the fix is ok.

File: lib/payload/cloak.vala [hunk 1]
```diff
@@ -99,23 +99,32 @@ namespace Frida {
 			listeners.add (readdir_listener);
 			interceptor.attach (readdir, readdir_listener);
 
-			var readdir64 = Gum.Module.find_export_by_name (libc, "readdir64");
-			if (readdir64 != null && readdir64 != readdir) {
+			/*
+			 * It seems that when using the musl CRT, the compiler incorrectly
+			 * determines that symbols with the same name, but a different
```
Comment (oleavr) [inline lib/payload/cloak.vala:104]: Hmm, perhaps the reason is that it has `#define readdir64` to something, and that's getting picked up here?

File: lib/payload/cloak.vala [hunk 2]
```diff
@@ -99,23 +99,32 @@ namespace Frida {
...
+			 * numeric suffix are the same symbol. Thus when the C code is
+			 * generate for this VALA, the symbols 'readdir' and 'readdir64'
+			 * collide resulting in a redefinition error. Hence the name has
+			 * been slightly changed.
+			 */
+			var readdir64_w = Gum.Module.find_export_by_name (libc,
+			"readdir64");
```
Comment (oleavr) [inline lib/payload/cloak.vala:111]: Lines should not be broken at 80 cols, but 140 cols in frida-core.

File: lib/payload/cloak.vala [hunk 3]
```diff
@@ -99,23 +99,32 @@ namespace Frida {
...
 			interceptor.attach (readdir_r, readdir_r_listener);
 
-			var readdir64_r = Gum.Module.find_export_by_name (libc, "readdir64_r");
-			if (readdir64_r != null && readdir64_r != readdir_r) {
+			var readdir64_r_w = Gum.Module.find_export_by_name (libc, "readdir64_r");
+			if (readdir64_r_w != null && readdir64_r_w != readdir_r) {
 				var listener = new ReadDirRListener (this, MODERN);
 				listeners.add (listener);
-				interceptor.attach (readdir64_r, listener);
+				interceptor.attach (readdir64_r_w, listener);
```
Comment (oleavr) [inline lib/payload/cloak.vala:127]: Let's use `_impl` as suffix instead.

File: lib/payload/cloak.vala (hunk 1 above)
Comment (WorksButNotTested) [reply lib/payload/cloak.vala]: Oh maybe, if they had something like:
#define readdir64 readdir

Then that could throw it off. Then the compiler warning message was showing the source file before it had been pre-processed and hence showing the name collision. I have updated the comment accordingly and fixed it to wrap at 140 cols.

File: lib/payload/cloak.vala [hunk 4]
```diff
@@ -99,23 +99,27 @@ namespace Frida {
...
 
-			var readdir64_r = Gum.Module.find_export_by_name (libc, "readdir64_r");
-			if (readdir64_r != null && readdir64_r != readdir_r) {
+			var readdir64_r_impl = Gum.Module.find_export_by_name (libc, "readdir64_r");
+			if (readdir64_r_impl != null && readdir64_r_impl != readdir_r) {
 				var listener = new ReadDirRListener (this, MODERN);
 				listeners.add (listener);
-				interceptor.attach (readdir64_r, listener);
+				interceptor.attach (readdir64_r_impl, listener);
 			}
```
Comment (oleavr) [inline lib/payload/cloak.vala:123]: I think we should also suffix the other names for consistency. I'm not sure we need the comment though, as it's generally a bad idea to name variables the same as standard functions.

**Comment by WorksButNotTested** [comment]
Made names consistent and removed comment.

**Pull request merged**